react/features/toolbar/actions.web.js

```diff
@@ -0,0 +1,291 @@
+/* @flow */
+
+import Recording from '../../../modules/UI/recording/Recording';
+import SideContainerToggler
+    from '../../../modules/UI/side_pannels/SideContainerToggler';
+import UIEvents from '../../../service/UI/UIEvents';
+import UIUtil from '../../../modules/UI/util/UIUtil';
+
+import {
+    clearToolbarTimeout,
+    setAlwaysVisibleToolbar,
+    setSubjectSlideIn,
+    setToolbarButton,
+    setToolbarTimeout,
+    setToolbarTimeoutNumber,
+    setToolbarVisibility,
+    toggleToolbarButton
+} from './actions.native';
+
+export * from './actions.native';
+
+declare var $: Function;
+declare var APP: Object;
+declare var config: Object;
+declare var interfaceConfig: Object;
+
+/**
+ * Checks whether desktop sharing is enabled and whether
+ * we have params to start automatically sharing.
+ *
+ * @returns {Function}
+ */
+export function checkAutoEnableDesktopSharing(): Function {
+    return () => {
+        // XXX Should use dispatcher to toggle screensharing but screensharing
+        // hasn't been React-ified yet.
+
+        if (UIUtil.isButtonEnabled('desktop')
+                && config.autoEnableDesktopSharing) {
+            APP.UI.eventEmitter.emit(UIEvents.TOGGLE_SCREENSHARING);
+        }
+    };
+}
+
+/**
+ * Docks/undocks toolbar based on its parameter.
+ *
+ * @param {boolean} isDock - True if dock, false otherwise.
+ * @returns {Function}
+ */
+export function dockToolbar(isDock: boolean): Function {
+    return (dispatch: Dispatch<*>, getState: Function) => {
+        if (interfaceConfig.filmStripOnly) {
+            return;
+        }
+
+        const state = getState();
+        const { toolbarTimeout, visible } = state['features/toolbar'];
+
+        if (isDock) {
+            // First make sure the toolbar is shown.
+            visible || dispatch(showToolbar());
+
+            dispatch(clearToolbarTimeout());
+        } else if (visible) {
+            dispatch(setToolbarTimeout(() => {
+                dispatch(hideToolbar());
+            }, toolbarTimeout));
+        } else {
+            dispatch(showToolbar());
+        }
+    };
+}
+
+/**
+ * Hides the toolbar.
+ *
+ * @param {boolean} force - True to force the hiding of the toolbar without
+ * caring about the extended toolbar side panels.
+ * @returns {Function}
+ */
+export function hideToolbar(force: boolean = false): Function {
+    return (dispatch: Dispatch<*>, getState: Function) => {
+        const state = getState();
+        const {
+            alwaysVisible,
+            isHovered,
+            toolbarTimeout
+        } = state['features/toolbar'];
+
+        if (alwaysVisible) {
+            return;
+        }
+
+        dispatch(clearToolbarTimeout());
+
+        if (!force && (isHovered
+                || APP.UI.isRingOverlayVisible()
+                || SideContainerToggler.isVisible())) {
```

Comment:
@BeatC, the formatting of the above does not reflect the order in which the boolean expressions will be evaluated. Please comply with our coding style (e.g. the consistency rule) and make sure to properly format the source code you submit as ready for PR review.